### PR TITLE
Redo persist push command for design V2

### PR DESCRIPTION
**Pull request by daneshk**
## Purpose

Following changes are included in this PR,
- Change the persist tool configurations structure in Ballerina.toml file.
- Change the init command to generate the new structure.
- Change the push command for the new design.
- Update persist module version and change the test cases. 
- Remove DB push-related classes from codecov ignore list

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/3949

## Checklist
- [x] Linked to an issue
- ~~[ ] Updated the specification~~
- ~~[ ] Updated the changelog~~
- [x] Added tests

### CODE REVIEW

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_1/Ballerina.toml [hunk 1]
```diff
@@ -6,3 +6,10 @@ distribution = "2201.3.0"
 
 [build-options]
 observabilityIncluded = false
+
+[persist.entities.storage.mysql]
+host = "localhost"
+port = 3307
+user = "root"
+password = "Test123#"
+database = "persist"
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_db_push_1/Ballerina.toml:15]:
```suggestion
database = "persist"

```

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_db_push_1/Ballerina.toml [hunk 2]
```diff
@@ -6,3 +6,10 @@ distribution = "2201.3.0"
 
 [build-options]
 observabilityIncluded = false
+
+[persist.entities.storage.mysql]
+host = "localhost"
+port = 3307
+user = "root"
+password = "Test123#"
+database = "persist"
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_db_push_1/Ballerina.toml:15]:
```suggestion
database = "persist"

```

File: persist-cli/src/main/java/io/ballerina/persist/PersistToolsConstants.java [hunk 3]
```diff
@@ -42,32 +41,18 @@ private PersistToolsConstants() {}
...
-    public static final String SQL_SCHEMA_FILE = "persist_db_scripts.sql";
-    public static final String UNIQUE_CONSTRAINTS = "uniqueConstraints";
+    public static final String PERSIST_CONFIG_PATTERN = "persist.%s.storage";
+    public static final String PERSIST_CONFIG_PATTERN_WITH_MYSQL = "persist.%s.storage.mysql";
+    public static final String SQL_SCHEMA_FILE = "%s_db_scripts.sql";
     public static final String PERSIST_DIRECTORY = "persist";
     public static final String GENERATED_DIRECTORY = "generated";
-    public static final String PERSIST_TOML_FILE = "Persist.toml";
-    public static final String PATH_BALLERINA_TOML = "Ballerina.toml";
+    public static final String BALLERINA_TOML_FILE = "Ballerina.toml";
```
Comment (niveathika) [inline persist-cli/src/main/java/io/ballerina/persist/PersistToolsConstants.java:55]: Hall we use the one from ballerina-lang ProjectAPIConstants?

File: persist-cli/src/main/java/io/ballerina/persist/cmd/Push.java [hunk 4]
```diff
@@ -115,73 +117,105 @@ public void execute() {
...
-                    Paths.get(projectPath.toString(), PERSIST_DIRECTORY));
-        } catch (ProjectException | BalException e) {
-            errStream.println("Error occurred while generating SQL schema. " + e.getMessage());
+        List<Path> schemaFilePaths;
+        try (Stream<Path> stream = Files.list(persistDir)) {
+            schemaFilePaths = stream.filter(file -> !Files.isDirectory(file))
+                    .filter(file -> file.toString().toLowerCase(Locale.ENGLISH).endsWith(".bal"))
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            errStream.println("Error while listing the persist schema files in persist directory. " + e.getMessage());
```
Comment (niveathika) [inline persist-cli/src/main/java/io/ballerina/persist/cmd/Push.java:133]:
```suggestion
            errStream.println("Error while listing the persist model definition files in persist directory. " + e.getMessage());
```
Shall we use "model definition" instad of schema? schem applies mostly for database related stuff

**Review by niveathika — APPROVED** [review]
Added minor comments, otherwise LGTM

File: persist-cli/src/main/java/io/ballerina/persist/PersistToolsConstants.java (hunk 3 above)
Comment (daneshk) [reply persist-cli/src/main/java/io/ballerina/persist/PersistToolsConstants.java]: Fixed in [83a51f2](https://github.com/ballerina-platform/persist-tools/pull/101/commits/83a51f236d9f746e16b46047d40281ecd05ba7fe)

**Review by niveathika — APPROVED** [review]

**Pull request merged**